lib/active_merchant/billing/gateways/monei.rb

```diff
@@ -0,0 +1,314 @@
+      #           :channel_id Channel ID
+      #           :login      User login
+      #           :pwd        User password
+      #
+      def initialize(options={})
+        requires!(options, :sender_id, :channel_id, :login, :pwd)
+        super
+      end
+
+      # Public: Performs purchase operation
+      #
+      # money       - Amount of purchase
+      # credit_card - Credit card
+      # options     - Hash containing purchase options
+      #               :order_id         Merchant created id for the purchase
+      #               :billing_address  Hash with billing address information
+      #               :description      Merchant created purchase description (optional)
+      #               :currency         Sale currency to override money object or default (optional)
+      #               :given            Customer given name to override card holder's (optional)
+      #               :family           Customer family name to override card holder's (optional)
+      #
+      # Returns Active Merchant response object
+      def purchase(money, credit_card, options)
+        execute_new_order(:purchase, money, credit_card, options)
+      end
+
+      # Public: Performs authorization operation
+      #
+      # money       - Amount to authorize
+      # credit_card - Credit card
+      # options     - Hash containing authorization options
+      #               :order_id         Merchant created id for the authorization
+      #               :billing_address  Hash with billing address information
+      #               :description      Merchant created authorization description (optional)
+      #               :currency         Sale currency to override money object or default (optional)
+      #               :given            Customer given name to override card holder's (optional)
+      #               :family           Customer family name to override card holder's (optional)
+      #
+      # Returns Active Merchant response object
+      def authorize(money, credit_card, options={})
+        execute_new_order(:authorize, money, credit_card, options)
+      end
+
+      # Public: Performs capture operation on previous authorization
+      #
+      # money         - Amount to capture
+      # authorization - Reference to previous authorization, obtained from response object returned by authorize
+      # options       - Hash containing capture options
+      #                 :order_id         Merchant created id for the authorization (optional)
+      #                 :description      Merchant created authorization description (optional)
+      #                 :currency         Sale currency to override money object or default (optional)
+      #
+      # Note: you should pass either order_id or description
+      #
+      # Returns Active Merchant response object
+      def capture(money, authorization, options={})
+        execute_authorization(:capture, money, authorization, options)
```

Comment:
These method names are a bit weird, `#execute_authorization` sounds like you're doing the initial auth, not acting on one.